cargo_wagon.rb

```diff
@@ -1,4 +1,28 @@
 require_relative 'wagon.rb'
 
 class CargoWagon < Wagon
+    attr_reader :volume, :free_space
+
+    def initialize(number, volume)
+        @volume = volume.to_f
+        super(number)
+        @free_space = volume
+    end
+
+    def take_up_space(volume)
+        self.free_space > volume ? self.free_space - volume : self.free_space = 0
```

Comment:
```suggestion
        free_space > volume ? free_space - volume : .free_space = 0
```
"селф" используется только при обращении к райтерам